contracts/stake-cw721/src/contract.rs

```diff
@@ -0,0 +1,1844 @@
+                .add_attribute("from", info.sender)
+                .add_attribute("token_id", token_id)
+                .add_attribute("claim_duration", "None")
+                .add_message(wasm_claim_rewards_msg)
+                .add_attribute(
+                    "reward_wallet",
+                    validated_reward_wallet.unwrap_or_else(|| DEFAULT_ADDR.clone()),
+                )
+                .add_attribute("reward_amount", amount_to_claim))
+        }
+
+        Some(duration) => {
+            let outstanding_claims = NFT_CLAIMS
+                .query_claims(deps.as_ref(), &info.sender)?
+                .nft_claims;
+            if outstanding_claims.len() >= MAX_CLAIMS as usize {
+                return Err(ContractError::TooManyClaims {});
+            }
+
+            NFT_CLAIMS.create_nft_claim(
+                deps.storage,
+                &info.sender,
+                token_id.clone(),
+                duration.after(&env.block),
+            )?;
+
+            if amount_to_claim.is_zero() || validated_reward_wallet == None {
+                return Ok(Response::new()
+                    .add_attribute("action", "unstake")
+                    .add_submessages(hook_msgs)
+                    .add_attribute("from", info.sender)
+                    .add_attribute("token_id", token_id)
+                    .add_attribute("claim_duration", format!("{}", duration)));
+            }
+
+            let outstanding_reward_claims = REWARD_CLAIMS
+                .query_claims(deps.as_ref(), &info.sender)?
+                .claims;
+            if outstanding_reward_claims.len() >= MAX_CLAIMS as usize {
+                return Err(ContractError::TooManyClaims {});
+            }
+
+            REWARD_CLAIMS.create_claim(
+                deps.storage,
+                &validated_reward_wallet
+                    .clone()
+                    .unwrap_or_else(|| DEFAULT_ADDR.clone()),
+                amount_to_claim,
+                duration.after(&env.block),
+            )?;
+
+            Ok(Response::new()
+                .add_attribute("action", "unstake")
+                .add_submessages(hook_msgs)
+                .add_attribute("from", info.sender)
+                .add_attribute("token_id", token_id)
+                .add_attribute("claim_duration", format!("{}", duration))
+                .add_attribute(
+                    "reward_wallet",
+                    validated_reward_wallet.unwrap_or_else(|| DEFAULT_ADDR.clone()),
+                )
+                .add_attribute("reward_amount", amount_to_claim))
+        }
+    }
+}
+
+pub fn execute_claim_nfts(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let nfts = NFT_CLAIMS.claim_nfts(deps.storage, &info.sender, &_env.block)?;
+    if nfts.is_empty() {
+        return Err(ContractError::NothingToClaim {});
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+    let mut msgs = Vec::new();
```

Comment:
Instead of making a mutable vector and pushing to it repeatedly how about doing this with iterators and `collect`? If we do it like that the compiler can do one allocation at the end instead of having to repeatedly resize the vector when we push.

Reply:
How do we handle the possible error from to_binary(&cw_transfer_msg)?